src/Servers/IIS/AspNetCoreModuleV2/CommonLib/Environment.cpp

```diff
@@ -231,7 +231,7 @@ bool Environment::CheckUpToDate(const std::wstring& source, const std::filesyste
             auto sourceInnerDirectory = std::filesystem::directory_entry(path);
             if (sourceInnerDirectory.path() != directoryToIgnore)
```

Comment:
I know this is unchanged, but why is this logic different than `CopyToDirectoryInner`? It seems like that logic works even if the original source in nested inside `directoryToIgnore` unlike this. Maybe that never happens, but it feels like we should use the same logic in both places if possible. We could do that in `main` though.

```cpp
            auto sourceInnerDirectory = path.path();

            if (sourceInnerDirectory.wstring().rfind(directoryToIgnore, 0) != 0)
```

And what is `auto sourceInnerDirectory = std::filesystem::directory_entry(path);` doing? Isn't path already a `directory_entry`?